src/main/java/com/librato/metrics/client/Sanitizer.java

```diff
@@ -37,6 +39,24 @@ public String apply(String name) {
         }
     };
 
+    @SuppressWarnings("unused")
+    public static final Sanitizer VALUE_SANITIZER = new Sanitizer() {
+        private final int lengthLimit = 256;
+        // "disallow anything that isn't a word char, dash, dot, colon, underscore, question mark or slash"
+        private final Pattern disallowedCharacters = Pattern.compile("[^-.:_?\\\\/\\w ]");
+
+        public String apply(final String value) {
+            if (value == null) {
+                return null;
+            }
+            final String sanitized = disallowedCharacters.matcher(value).replaceAll("");
+            if (sanitized.length() > lengthLimit) {
+                return sanitized.substring(sanitized.length() - lengthLimit, sanitized.length());
```

Comment:
I see that you're just keeping the existing behavior, but I'm really curious why we truncate from the front rather than the back of the string.

Reply:
I really have no idea. The test I change to remove the randomness tests this explicitly. The test string it uses looks like a package name of sorts, so I suppose it's about truncating the bits that are repetitive and keeping the parts that are meaningful.

Reply:
yeah, that makes sense!